### PR TITLE
Added documentation for new default workflow templates

### PR DESCRIPTION
**Pull request by amitgalitz**
### Description
Added documentation for new reindex template, this is an additional part of this PR: https://github.com/opensearch-project/documentation-website/pull/7271 along with two new templates that we have already released in managed service that weren't a part of doc until now. 

Two other pretrained templates were added as part of this meta issue: https://github.com/opensearch-project/flow-framework/issues/666

### Issues Resolved
Resolves https://github.com/opensearch-project/documentation-website/issues/7243
### Version
2.15

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by amitgalitz** [comment]
@owaiskazi19 @dbwiddis can you take a look?

**Review by joshpalis — APPROVED** [review]

**Comment by hdhalter** [comment]
Confirmed this is ready for doc review, @kolchfa-aws. Thanks!

File: _automating-configurations/workflow-templates.md [hunk 1]
```diff
@@ -138,5 +138,8 @@ The following table lists the supported workflow templates. To use a workflow te
...
 | `conversational_search_with_llm_deploy`  | Deploys a large language model (LLM) (by default, Cohere Chat) and configures a search pipeline with a `retrieval_augmented_generation` processor for [conversational search]({{site.url}}{{site.baseurl}}/search-plugins/conversational-search/).  | `create_connector.credential.key`  |[Defaults](https://github.com/opensearch-project/flow-framework/blob/2.13/src/main/resources/defaults/conversational-search-defaults.json) |
+| `semantic_search_with_reindex`  | Configures [semantic search]({{site.url}}{{site.baseurl}}/search-plugins/semantic-search/) with a newly deployed Cohere embedding model with a reindex a source index to a newly configured knn index. You must provide the API key for the Cohere model along with the source index to be reindexed. | `create_connector.credential.key`, `reindex.source_index`|[Defaults](https://github.com/opensearch-project/flow-framework/blob/main/src/main/resources/defaults/semantic-search-with-reindex-defaults.json) |
```
Comment (kolchfa-aws) [inline _automating-configurations/workflow-templates.md:141]:
```suggestion
| `semantic_search_with_reindex`  | Configures [semantic search]({{site.url}}{{site.baseurl}}/search-plugins/semantic-search/) with a newly deployed Cohere embedding model. The model is configured to reindex a source index to a newly configured k-NN index. You must provide the API key for the Cohere model along with the source index to be reindexed. | `create_connector.credential.key`, `reindex.source_index`|[Defaults](https://github.com/opensearch-project/flow-framework/blob/main/src/main/resources/defaults/semantic-search-with-reindex-defaults.json) |
```

File: _automating-configurations/workflow-templates.md [hunk 2]
```diff
@@ -138,5 +138,8 @@ The following table lists the supported workflow templates. To use a workflow te
...
+| `semantic_search_with_local_model`  | Configures [semantic search]({{site.url}}{{site.baseurl}}/search-plugins/semantic-search/) and deploys a pretrained model (`msmarco-distilbert-base-tas-b`). Adds a [`query_enricher`]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/neural-query-enricher/) search processor that sets a default model ID for neural queries and creates a linked k-NN index called `my-nlp-index`. You must provide the API key for the Cohere model.  | None  | [Defaults](https://github.com/opensearch-project/flow-framework/blob/main/src/main/resources/defaults/semantic-search-with-local-model-defaults.json) |
+| `hybrid_search_with_local_model`  | Configures [hybrid search]({{site.url}}{{site.baseurl}}/search-plugins/hybrid-search/): <br> - through deploying a pretrained model (`msmarco-distilbert-base-tas-b`). Creates an ingest pipeline, a k-NN index, and a search pipeline with a normalization_processor.  | None  | [Defaults](https://github.com/opensearch-project/flow-framework/blob/main/src/main/resources/defaults/hybrid-search-with-local-model-defaults.json) |
```
Comment (kolchfa-aws) [inline _automating-configurations/workflow-templates.md:143]:
```suggestion
| `hybrid_search_with_local_model`  | Configures [hybrid search]({{site.url}}{{site.baseurl}}/search-plugins/hybrid-search/) through deploying a pretrained model (`msmarco-distilbert-base-tas-b`). Creates an ingest pipeline, a k-NN index, and a search pipeline with a `normalization_processor`.  | None  | [Defaults](https://github.com/opensearch-project/flow-framework/blob/main/src/main/resources/defaults/hybrid-search-with-local-model-defaults.json) |
```

**Review by kolchfa-aws — APPROVED** [review]
Thank you, @amitgalitz! A couple of quick comments, and we'll be ready to move this PR to editorial review.

File: _automating-configurations/workflow-templates.md [hunk 3]
```diff
@@ -138,5 +138,8 @@ The following table lists the supported workflow templates. To use a workflow te
...
+| `semantic_search_with_local_model`  | Configures [semantic search]({{site.url}}{{site.baseurl}}/search-plugins/semantic-search/) and deploys a pretrained model (`msmarco-distilbert-base-tas-b`). Adds a [`query_enricher`]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/neural-query-enricher/) search processor that sets a default model ID for neural queries and creates a linked k-NN index called `my-nlp-index`. You must provide the API key for the Cohere model.  | None  | [Defaults](https://github.com/opensearch-project/flow-framework/blob/main/src/main/resources/defaults/semantic-search-with-local-model-defaults.json) |
+| `hybrid_search_with_local_model`  | Configures [hybrid search]({{site.url}}{{site.baseurl}}/search-plugins/hybrid-search/) through deploying a pretrained model (`msmarco-distilbert-base-tas-b`). Creates an ingest pipeline, a k-NN index, and a search pipeline with a `normalization_processor`.  | None  | [Defaults](https://github.com/opensearch-project/flow-framework/blob/main/src/main/resources/defaults/hybrid-search-with-local-model-defaults.json) |
```
Comment (natebower) [inline _automating-configurations/workflow-templates.md:143]: "through deploying a pretrained model" => either "and deploys a pretrained model" (to match the preceding row) or "by deploying a pretrained model".

Comment (kolchfa-aws) [inline _automating-configurations/workflow-templates.md:143]:
```suggestion
| `hybrid_search_with_local_model`  | Configures [hybrid search]({{site.url}}{{site.baseurl}}/search-plugins/hybrid-search/) and deploys a pretrained model (`msmarco-distilbert-base-tas-b`). Creates an ingest pipeline, a k-NN index, and a search pipeline with a `normalization_processor`.  | None  | [Defaults](https://github.com/opensearch-project/flow-framework/blob/main/src/main/resources/defaults/hybrid-search-with-local-model-defaults.json) |
```

**Pull request merged**